libaleth-interpreter/VMCalls.cpp

```diff
@@ -190,7 +190,13 @@ void VM::caseCall()
 
 bool VM::caseCallSetup(evmc_message& o_msg, bytesRef& o_output)
 {
-    m_runGas = m_rev >= EVMC_TANGERINE_WHISTLE ? 700 : 40;
+    evmc_address const destination = toEvmC(asAddress(m_SP[1]));
+    
+    // Check for call-to-self (eip1380) and adjust gas accordingly
+    if (fromEvmC(m_message->destination) == fromEvmC(destination) && m_rev >= EVMC_BERLIN)
```

Comment:
You need to have the cheap `m_rev >= EVMC_BERLIN` check first.

Reply:
@chfast Ah good point, comparing addresses is expensive since you're comparing 32 byte boost multiprecision values which are also first converted from big endian, instead of an EVMC_REVISION which is an enum (4 bytes).